src/components/SnapshotDetails/tabs/__tests__/SnapshotOverview.spec.tsx

```diff
@@ -0,0 +1,169 @@
+import { screen } from '@testing-library/react';
+import { usePipelineRunV2 } from '~/hooks/usePipelineRunsV2';
+import { useScanResults } from '~/hooks/useScanResults';
+import { useSnapshot } from '~/hooks/useSnapshots';
+import { mockUseNamespaceHook } from '~/unit-test-utils/mock-namespace';
+import { renderWithQueryClientAndRouter } from '~/utils/test-utils';
+import SnapshotOverview from '../SnapshotOverview';
+
+// Mock router to turn Link into a simple anchor and control params
+jest.mock('react-router-dom', () => {
+  const actual = jest.requireActual('react-router-dom');
+  return {
+    ...actual,
+    Link: (props: {
+      to: string;
+      'data-test': string;
+      title: string;
+      children: React.ReactNode;
+    }) => (
+      <a href={props.to} data-test={props['data-test']} title={props.title}>
+        {props.children}
+      </a>
+    ),
+    useParams: jest.fn(() => ({ snapshotName: 'snap-1' })),
+  };
+});
+
+// Mock snapshot hook
+jest.mock('~/hooks/useSnapshots', () => ({
+  useSnapshot: jest.fn(),
+}));
+
+// Mock pipelinerun hook
+jest.mock('~/hooks/usePipelineRunsV2', () => ({
+  usePipelineRunV2: jest.fn(),
+}));
+
+// Mock scan results hook
+jest.mock('~/hooks/useScanResults', () => ({
+  useScanResults: jest.fn(),
+}));
+
+// Mock scroll hook to be a no-op
+jest.mock('~/hooks/useScrollToHash', () => ({
+  useScrollToHash: jest.fn(),
+}));
+
+// Mock commit util to return a stable commit object
+jest.mock('~/utils/commits-utils', () => ({
+  createCommitObjectFromPLR: jest.fn(() => ({
+    sha: 'abc5b3ad0a2c16726523b12cf3b8f0365be33566',
+    shaTitle: 'abc123',
+    displayName: 'Commit title',
+    gitProvider: 'github',
+    shaURL: 'https://example.com/commit/abc5b3ad0a2c16726523b12cf3b8f0365be33566',
+  })),
+  ...jest.requireActual('~/utils/commits-utils'),
+  getCommitShortName: jest.fn((sha: string) => sha.slice(0, 7)),
+}));
+
+const useSnapshotMock = useSnapshot as jest.Mock;
+const usePipelineRunV2Mock = usePipelineRunV2 as jest.Mock;
+const useScanResultsMock = useScanResults as jest.Mock;
+const useNamespaceMock = mockUseNamespaceHook('test-ns');
+
+const baseSnapshot = {
+  metadata: {
+    name: 'snap-1',
+    namespace: 'test-ns',
+    creationTimestamp: '2024-01-01T00:00:00Z',
+    labels: {
+      'appstudio.openshift.io/build-pipeline': 'build-plr-1',
+    },
+  },
+  spec: {
+    application: 'app-1',
+    components: [{ name: 'comp-a' }, { name: 'comp-b' }],
+  },
+};
+
```

Comment:
we have some testing data in src/__data__/pipelinerun-data.ts.
If we reuse some, i guess the code could be shorter.